pkg/controller/persistentvolumelabel/pvlcontroller.go

```diff
@@ -0,0 +1,421 @@
+	pvlc.mutex.Lock()
+	defer pvlc.mutex.Unlock()
+
+	if pvlc.gceCloudProvider == nil {
+		gceCloudProvider, ok := pvlc.cloud.(*gce.GCECloud)
+		if !ok {
+			// GetCloudProvider has gone very wrong
+			return nil, fmt.Errorf("error retrieving GCE cloud provider")
+		}
+		pvlc.gceCloudProvider = gceCloudProvider
+	}
+	return pvlc.gceCloudProvider, nil
+}
+
+func (pvlc *PersistentVolumeLabelController) updateVolume(volume *v1.PersistentVolume) error {
+	//func (pvlc *PersistentVolumeLabelController) updateVolume(labels map[string]string) error {
+	glog.V(4).Infof("updating PersistentVolume %s", volume.Name)
+	err := clientretry.RetryOnConflict(updatePVLabelsBackoff, func() error {
+		curVol, err := pvlc.kubeClient.Core().PersistentVolumes().Get(volume.Name, metav1.GetOptions{IncludeUninitialized: true})
+		if err != nil {
+			return err
+		}
+		oldData, err := json.Marshal(curVol)
+		if err != nil {
+			return fmt.Errorf("failed to marshal old persistentvolume %#v for persistentvolume %q: %v", curVol, volume.Name, err)
+		}
+
+		objCopy, err := scheme.Scheme.DeepCopy(curVol)
+		if err != nil {
+			return fmt.Errorf("failed to copy persistentvolume object %#v: %v", curVol, err)
+		}
+		newVolume, ok := (objCopy).(*v1.PersistentVolume)
+		if !ok {
+			return fmt.Errorf("failed to cast copy into persistentvolume object %#v: %v", newVolume, err)
+		}
+		newVolume.Labels = volume.Labels
+		pvlc.removeInitializer(newVolume)
+
+		newData, err := json.Marshal(newVolume)
+		if err != nil {
+			return fmt.Errorf("failed to marshal new persistentvolume %#v for persistentvolume %q: %v", newVolume, volume.Name, err)
+		}
+
+		patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldData, newData, v1.PersistentVolume{})
+		if err != nil {
+			return fmt.Errorf("failed to create patch for persistentvolume %q: %v", volume.Name, err)
+		}
+
+		_, err = pvlc.kubeClient.Core().PersistentVolumes().Patch(string(volume.Name), types.StrategicMergePatchType, patchBytes)
+		return err
+	})
+	if err == nil {
+		glog.V(4).Infof("updated PersistentVolume %s", volume.Name)
+	}
+
+	return err
+}
+
+func (pvlc *PersistentVolumeLabelController) addInitializationConfig() {
+	pvlInitializer := v1alpha1.Initializer{
+		Name: initializerName,
+		Rules: []v1alpha1.Rule{
+			{
+				APIGroups:   []string{"*"},
+				APIVersions: []string{"*"},
+				Resources:   []string{"persistentvolumes"},
+			},
+		},
+	}
+	pvlInitConfig := v1alpha1.InitializerConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: initialzerConfigName,
+		},
+		Initializers: []v1alpha1.Initializer{pvlInitializer},
+	}
+
+	existingInitConfig, err := pvlc.kubeClient.AdmissionregistrationV1alpha1().InitializerConfigurations().Get(initialzerConfigName, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// InitializerConfig wasn't found
+			_, err = pvlc.kubeClient.AdmissionregistrationV1alpha1().InitializerConfigurations().Create(&pvlInitConfig)
```

Comment:
As far as I know, InitializerConfigurations have a unique name and each one can act upon different sets of resources.  Since this is internal functionality to kube, imo it makes sense for the name and the InitializerConfiguration to be owned by this process.  An admin can always create a different InitializerConfiguration if they want to do something with volume labels.